cli/flox-activations/src/cli/activate/activate_script_builder.rs

```diff
@@ -12,69 +18,173 @@ pub const FLOX_SERVICES_SOCKET_VAR: &str = "_FLOX_SERVICES_SOCKET";
+        // This is used by set-prompt script, and tcsh in particular does not
+        // tolerate references to undefined variables.
+        (
+            FLOX_PROMPT_ENVIRONMENTS_VAR,
+            context.flox_prompt_environments,
+        ),
+        ("_FLOX_SET_PROMPT", context.set_prompt.to_string()),
+        (
+            "_FLOX_ACTIVATE_STORE_PATH",
+            context.flox_activate_store_path,
+        ),
         (
             // TODO: we should probably figure out a more consistent way to
             // pass this since it's also passed for `flox build`
             FLOX_RUNTIME_DIR_VAR,
-            data.flox_runtime_dir,
+            context.flox_runtime_dir,
         ),
-        ("_FLOX_ENV_CUDA_DETECTION", data.flox_env_cuda_detection),
+        ("_FLOX_ENV_CUDA_DETECTION", context.flox_env_cuda_detection),
         (
             FLOX_ACTIVATE_START_SERVICES_VAR,
-            data.flox_activate_start_services.to_string(),
+            context.flox_activate_start_services.to_string(),
         ),
     ]);
-    if let Some(log_dir) = data.flox_env_log_dir.as_ref() {
+    if let Some(log_dir) = context.flox_env_log_dir.as_ref() {
         exports.insert(FLOX_ENV_LOG_DIR_VAR, log_dir.clone());
     }
-    if let Some(socket_path) = data.flox_services_socket.as_ref() {
+    if let Some(socket_path) = context.flox_services_socket.as_ref() {
         exports.insert(FLOX_SERVICES_SOCKET_VAR, socket_path.clone());
     }
-    if let Some(services_to_start) = data.flox_services_to_start {
+    if let Some(services_to_start) = context.flox_services_to_start {
         exports.insert(FLOX_SERVICES_TO_START_VAR, services_to_start);
     }
 
     exports.extend(default_nix_env_vars());
 
     command.envs(exports);
 
-    command.arg("--env").arg(&data.env);
-    if let Some(env_project) = data.env_project.as_ref() {
+    if let Some(env_project) = context.env_project.as_ref() {
         command
             .arg("--env-project")
             .arg(env_project.to_string_lossy().to_string());
     }
     command
         .arg("--env-cache")
-        .arg(data.env_cache.to_string_lossy().to_string());
-    command.arg("--env-description").arg(data.env_description);
+        .arg(context.env_cache.to_string_lossy().to_string());
+    command
+        .arg("--env-description")
+        .arg(context.env_description);
 
     // Pass down the activation mode
-    command.arg("--mode").arg(data.mode);
+    command.arg("--mode").arg(context.mode);
 
-    if let Some(watchdog_bin) = data.watchdog_bin.as_ref() {
+    if let Some(watchdog_bin) = context.watchdog_bin.as_ref() {
         command
             .arg("--watchdog")
             .arg(watchdog_bin.to_string_lossy().to_string());
     }
 
-    command.arg("--shell").arg(data.shell.exe_path());
+    command.arg("--shell").arg(context.shell.exe_path());
+}
+
+/// Prior to the refactor, these variables were exported in the activate script
+// TODO: we still use std::env::var in this function,
```

Comment:
Suggestion: put them on `VarsFromEnvironment`, don't drop it. This type will make it easier to test without needing an actual process environment.

Reply:
Yeah I think that will probably be best, just wanting to leave as a TODO for now